src/main/java/org/broadinstitute/hellbender/tools/copynumber/PreprocessIntervals.java

```diff
@@ -0,0 +1,128 @@
+package org.broadinstitute.hellbender.tools.copynumber;
+
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.util.Interval;
+import htsjdk.samtools.util.IntervalList;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.barclay.help.DocumentedFeature;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.engine.GATKTool;
+import org.broadinstitute.hellbender.utils.IntervalUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.cmdline.programgroups.CopyNumberProgramGroup;
+import org.apache.commons.math3.util.FastMath;
+import java.io.File;
+import java.util.List;
+
+/**
+ * This tool takes in intervals via the standard arguments of {@link GATKTool} preprocesses them and creates bins that
+ * cover the processed intervals. The intervals are first padded, then they are uniquified and the overlapping intervals
+ * are merged. Finally, the bins are created.
+ *
+ * <p>Standard GATK engine arguments include -L and -XL. For example, for the -L argument, the tool accepts gatk-style
+ * intervals (.list or .intervals), BED files and VCF files. If no intervals are given, each contig will be covered
+ * by a single interval (whole genome sequencing). </p>
+ *
+ * <p>Using the -P flag, the user can specify the amount of padding (in bp) added to each side of the intervals.
+ * This padding is in addition to the padding added by the -ip (or --interval_padding) argument of IntervalArgumentCollection.
+ * We encourage using only the -P flag.
+ *
+ * <p>The user can also specify the length of the bins (in bp) using the -BL option. If this is not commensurate with
+ * the length of the padded intervals, then the last bin will be of different size than the others. </p>
+ *
+ * <h3>Example</h3>
+ * <pre>
+ * gatk-launch --javaOptions "-Xmx4g" PreprocessIntervals \
+ *   -R ref_fasta.fa \
+ *   -L intervals.list \
+ *   -BL 10000 \
+ *   -P 500 \
+ *   -O interval-files
+ * </pre>
+ *
+ * @author Marton Kanasz-Nagy &lt;mkanaszn@broadinstitute.org&gt;
+ */
+@CommandLineProgramProperties(
+        summary = "Preprocess intervals given by the user and then create bins that cover them. "
+                + "The intervals are first padded, the overlapping ones are merged and bins are created that cover the intervals. "
+                + "If the bin size is incommensurate with the length of the interval, the last bin of the interval might be of different size than the the rest of the bins."
```

Comment:
`length of an interval, the last bin of that interval will be of different size`

Reply:
OK, thank you.